projector-server/src/main/kotlin/org/jetbrains/projector/server/service/ProjectorDrawEventQueue.kt

```diff
@@ -87,7 +82,10 @@ class ProjectorDrawEventQueue private constructor(val target: ServerDrawCommands
     }
 
     private fun build() {
-      commands.add(events)
+      commands.add(target to events)
+      if (events.asReversed().none { it is ServerWindowPaintEvent }) {
```

Comment:
Could you clarify the purpose of `asReversed` here?

Reply:
Optimization: events have a `ServerWindowPaintEvent` at the end like this:

`[ServerWindowStateEvent, some more ServerWindowStateEvent, ServerWindowPaintEvent]`

So if we start checking from the end, the `none` call will return after the first check normally. I think I should add a comment here to clarify that since it just took me some time to remember:

```suggestion
      // asReversed: optimization. We expect that the last element is ServerWindowPaintEvent most of the time
      if (events.asReversed().none { it is ServerWindowPaintEvent }) {
```

WDYT?

Reply:
Yeah, I think the comment won't be excessive, please add it